apps/charge-api-service/src/consensus-api/consensus-api.controller.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  Body,
+  Controller,
+  Get,
+  Post,
+  UseGuards,
+  UseInterceptors
+} from '@nestjs/common'
+import { IsValidPublicApiKeyGuard } from '@app/api-service/api-keys/guards/is-valid-public-api-key.guard'
+import { ConsensusApiService } from '@app/api-service/consensus-api/consensus-api.service'
+import { DelegatedAmountsDto } from '@app/network-service/consensus/dto/consensus.dto'
+import { CacheInterceptor } from '@nestjs/cache-manager'
+
+@UseGuards(IsValidPublicApiKeyGuard)
+@Controller('v0/consensus')
+@UseInterceptors(CacheInterceptor)
+export class ConsensusApiController {
+  constructor (
+    private readonly consensusApiService: ConsensusApiService
+  ) { }
+
+  @Get('validators')
+  getValidators () {
+    return this.consensusApiService.getValidators()
+  }
+
+  @Post('delegated_amounts')
+  getDelegatedAmounts (@Body() delegatedAmountsDto: DelegatedAmountsDto) {
+    return this.consensusApiService.getDelegatedAmounts(delegatedAmountsDto)
+  }
+
+  @Post('delegate')
+  delegate (@Body() data: { validator: string }) {
+    return this.consensusApiService.delegate(data.validator)
+  }
+
+  @Post('withdraw')
+  withdraw (@Body() data: { validator: string, amount: string }) {
```

Comment:
@LiorAgnin both `delegate` and `withdraw` requires signer to execute write call on smart contract, how both of these functions are working?